cmake_unofficial/CMakeLists.txt

```diff
@@ -1,40 +1,108 @@
-cmake_minimum_required(VERSION 2.6)
-cmake_policy(VERSION 2.6)
+# To the extent possible under law, the author(s) have dedicated all
+# copyright and related and neighboring rights to this software to
+# the public domain worldwide. This software is distributed without
+# any warranty.
+#
+# For details, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 
-project(xxhash)
+set(XXHASH_TOP_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/..")
 
-set(XXHASH_LIB_VERSION "0.42.0")
-set(XXHASH_LIB_SOVERSION "0")
+# Parse version information
+file(STRINGS "${XXHASH_TOP_SOURCE_DIR}/xxhash.h" XXHASH_VERSION_MAJOR REGEX "^#define XXH_VERSION_MAJOR +([0-9]+) *$")
+string(REGEX REPLACE "^#define XXH_VERSION_MAJOR +([0-9]+) *$" "\\1" XXHASH_VERSION_MAJOR "${XXHASH_VERSION_MAJOR}")
+file(STRINGS "${XXHASH_TOP_SOURCE_DIR}/xxhash.h" XXHASH_VERSION_MINOR REGEX "^#define XXH_VERSION_MINOR +([0-9]+) *$")
+string(REGEX REPLACE "^#define XXH_VERSION_MINOR +([0-9]+) *$" "\\1" XXHASH_VERSION_MINOR "${XXHASH_VERSION_MINOR}")
+file(STRINGS "${XXHASH_TOP_SOURCE_DIR}/xxhash.h" XXHASH_VERSION_RELEASE REGEX "^#define XXH_VERSION_RELEASE +([0-9]+) *$")
```

Comment:
Thanks for reviewing @ChrisKitching !

But in this case, CMake must consume the header.

We must avoid to rely on CMake for fundamental purpose.  Since vesion number is one of the vital part of xxHash, we've been keeping it as simple as possible : plain code.
And it seems it's good decision.  Because there are a lot of projects which just include/copy xxHash code and simply it works.

Please note that so far we're also avoiding other (good but) complex and environment dependent things such as `autoconf` to keep plain code working well.